Gordon360/Controllers/HousingController.cs

```diff
@@ -229,4 +235,463 @@ public ActionResult<ApartmentApplicationViewModel[]> GetAllApartmentApplication(
+    [HttpPost("roomrange")]
+    [StateYourBusiness(operation = Operation.ADD, resource = Resource.HOUSING_ROOM_RANGE)]
+    public async Task<ActionResult<Hall_Assignment_Ranges>> CreateRoomRange([FromBody] HallAssignmentRangeViewModel model)
+    {
+        try
+        {
+            var newRange = await housingService.CreateRoomRangeAsync(model);
+            return Created("Room range created successfully.", newRange);
+        }
+        catch (InvalidOperationException ex)
+        {
+            return BadRequest(ex.Message);
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, $"Internal server error: {ex.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Retrieves all room ranges.
+    /// </summary>
+    /// <returns>A list of room ranges.</returns>
+    [HttpGet("roomrange/all")]
+    [StateYourBusiness(operation = Operation.READ_ALL, resource = Resource.HOUSING_ROOM_RANGE)]
+    public async Task<ActionResult<List<HallAssignmentRangeViewModel>>> GetAllRoomRanges()
+    {
+        try
+        {
+            var roomRanges = await housingService.GetAllRoomRangesAsync();
+            return Ok(roomRanges);
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, $"Internal server error: {ex.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Deletes a Room Range
+    /// </summary>
+    /// <param name="rangeId">The ID of the room range to delete</param>
+    /// <returns> Returns if completed</returns>
+    [HttpDelete("roomrange/{rangeId}")]
+    [StateYourBusiness(operation = Operation.DELETE, resource = Resource.HOUSING_ROOM_RANGE)]
+    public async Task<IActionResult> DeleteRoomRange(int rangeId)
+    {
+        try
+        {
+            var success = await housingService.DeleteRoomRangeAsync(rangeId);
+            if (success)
+            {
+                return Ok(new { message = "Room range deleted successfully." });
+            }
+            else
+            {
+                return BadRequest(new { message = "Failed to delete the room range." });
+            }
+        }
+        catch (ResourceNotFoundException ex)
+        {
+            return NotFound(new { message = ex.ExceptionMessage });
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, $"Internal server error: {ex.Message}");
+        }
+    }
+
+
+    /// <summary>
+    /// Assigns an RA to a room range if no RA is currently assigned
+    /// </summary>
+    /// <param name="range_Id">The ID of the room range</param>
+    /// <param name="ra_Id">The ID of the RA to assign</param>
+    /// <returns>The created RA_Assigned_Ranges object</returns>
+    [HttpPost("roomrange/assign-ra")]
+    [StateYourBusiness(operation = Operation.ADD, resource = Resource.HOUSING_RA_ASSIGNMENT)]
+    public async Task<ActionResult<RA_Assigned_Ranges>> AssignRaToRoomRange([FromBody] RA_AssignmentViewModel model)
```

Comment:
This route would idiomatically be `POST /roomranges/{roomRangeId}/assignment`, or else just `POST /roomrangeassignments` if we treat assignments as a stand-alone resource.

Reply:
Added to change branch